src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonObject.cs

```diff
@@ -32,7 +32,7 @@ public sealed partial class JsonObject : JsonNode
         /// </summary>
         /// <param name="properties">The properties to be added.</param>
         /// <param name="options">Options to control the behavior.</param>
-        public JsonObject(IEnumerable<KeyValuePair<string, JsonNode?>> properties, JsonNodeOptions? options = null)
+        public JsonObject(IEnumerable<KeyValuePair<string, JsonNode?>> properties, JsonNodeOptions? options = null) : this(options)
```

Comment:
Is there a test that can be added that would have failed before without this and will now succeed?

Reply:
If I am understanding the issue correctly. Then JsonNodeOptions doesn't get set when calling the constructor.
The proposed fix makes sure to set the JsonNodeOptions through the constructor JsonObject(JsonNodeOptions? options = null).

Suggestion: Add this test to runtime/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonNode/JsonObjectTests.cs 

```CS
[Fact]
public void NullJsonNodeOptionsCheck()
{
	JsonNodeOptions options = new JsonNodeOptions();
	IEnumerable<KeyValuePair<string, JsonNode?>> props = new List<KeyValuePair<string, JsonNode?>>();
	options.PropertyNameCaseInsensitive = true;
	JsonObject JO = new JsonObject(props, options);

	Assert.NotNull(JO.Options);
	Assert.True(JO.Options.Value.PropertyNameCaseInsensitive);
}
```

Currently this test fails.